euphonic/cli/intensity_map.py

```diff
@@ -0,0 +1,183 @@
+import argparse
+from typing import List, Optional, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import euphonic
+from euphonic import ureg
+import euphonic.plot
+from euphonic.util import get_qpoint_labels
+from .utils import (_bands_from_force_constants,
+                    get_args, _get_energy_unit, _get_q_distance,
+                    load_data_from_file)
+
+
+_spectrum_choices = ('dos', 'coherent')
+
+
+def get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('filename', type=str,
+                        help=('Phonon data file. This should contain force '
+                              'constants or band data. Force constants '
+                              'formats: .yaml, .hdf5 (Phonopy); .castep_bin , '
+                              '.check (Castep); .json (Euphonic). [A band '
+                              'structure path will be obtained using Seekpath.'
+                              '] Band data formats: .phonon (Castep); .json '
+                              '(Euphonic)'
+                              ))
+    parser.add_argument('--weights', '-w', default='dos',
+                        choices=_spectrum_choices,
+                        help=('Spectral weights to plot: phonon DOS or '
+                              'coherent inelastic neutron scattering.'))
+    parser.add_argument('--ebins', type=int, default=200,
+                        help='Number of energy bins on y-axis')
+    parser.add_argument('--e-min', type=float, default=None, dest='e_min',
+                        help='Energy range minimum in ENERGY_UNIT')
+    parser.add_argument('--e-max', type=float, default=None, dest='e_max',
+                        help='Energy range maximum in ENERGY_UNIT')
+    parser.add_argument('--v-min', type=float, default=None, dest='v_min',
+                        help='Minimum of data range for colormap.')
+    parser.add_argument('--v-max', type=float, default=None, dest='v_max',
+                        help='Maximum of data range for colormap.')
+    parser.add_argument('--length-unit', type=str, default='angstrom',
+                        dest='length_unit',
+                        help=('Length units; these will be inverted to obtain'
+                              'units of distance between q-points (e.g. "bohr"'
```

Comment:
Sorry very nitpicky but I just noticed there's no space between 'obtain' and 'units' (fixed in dispersion but not here)

Reply:
done!